pkgs/build-support/docker/store-path-to-layer.sh

```diff
@@ -5,16 +5,29 @@ set -eu
 layerNumber=$1
 shift
 
+storePath="$1"
+shift
+
 layerPath="./layers/$layerNumber"
-echo "Creating layer #$layerNumber for $@"
+echo "Creating layer #$layerNumber for $storePath"
 
 mkdir -p "$layerPath"
-tar --no-recursion -rf "$layerPath/layer.tar" \
+
+# make sure /nix and /nix/store appear first in the archive
+mkdir -p nix/store
+tar -cf "$layerPath/layer.tar"  \
     --mtime="@$SOURCE_DATE_EPOCH" \
-    --owner=0 --group=0 /nix /nix/store
-tar -rpf "$layerPath/layer.tar" --hard-dereference --sort=name \
+    --owner=0 --group=0 \
+    --transform='s,nix,/nix,' \
+    nix
+
+n=$(basename "$storePath")
+tar -C /nix/store -rpf "$layerPath/layer.tar" \
+    --hard-dereference --sort=name \
     --mtime="@$SOURCE_DATE_EPOCH" \
-    --owner=0 --group=0 "$@"
+    --owner=0 --group=0 \
+    --transform="s,$n,/nix/store/$n," \
```

Comment:
I don't understand why this change would be required. If I remember well, tar doesn't create entries for the directories `/nix` and `/nix/store` in the tar archive. Since they are not explicitly created by tar, it doesn't read them and I don't see any issues here regarding these entries.

Reply:
You are correct that `tar` does not create entries for `/nix` and `/nix/store`. But even though it doesn't create those entries, it does still check if they are modified while it's running. I was able to force `tar` to fail in this way by running a `nix build` of a bunch of packages and at the same time running 

```
tar -cf test.tar /nix/store/vs7zapqwngylvpydpv5dg3gk1va8dj8w-bash-interactive-4.4-p23
```

repeatedly.

Reply:
Ok. Nice stress test and thx for your explanation.